cabal-install/src/Distribution/Client/Manpage.hs

```diff
@@ -78,9 +78,11 @@ manpageCmd pname commands flags
 
         unless (ec1 == ExitSuccess) $ exitWith ec1
 
-        pager <- fromMaybe "less" <$> lookupEnv "PAGER"
-        -- 'less' is borked with color sequences otherwise
-        let pagerArgs = if takeFileName pager == "less" then ["-R"] else []
+        pagerAndArgs <- fromMaybe "less -R" <$> lookupEnv "PAGER"
+        -- 'less' is borked with color sequences otherwise, hence -R
+        (pager, pagerArgs) <- case words pagerAndArgs of
```

Comment:
Note that GNU man also supports some shell escaping/quoting rules here. The question is, of course, how many rely on this feature.

Reply:
Hmm, do you have an example in mind? If you want to experiment, you can download the cabal binary built with this patch [on the CI page](https://github.com/haskell/cabal/actions/runs/2895724823) (very bottom of it).

I don’t think compatibility with GNU man (or any other implementation of the format) was necessarily the goal of `cabal man` — only compatibility with the format. But we certainly want to support most widely-used scenarios.

Reply:
In theory you can have spaces in there:

```console
> echo "#!/bin/sh"\n'exec less "$@"' > weird\ pager
> env PAGER=$HOME/weird\\\ pager man less
> env PAGER=$HOME/\"weird\ pager\" man less
```

This is just something to note, OpenBSD's man implementation for example doesn't support this, so I doubt it's something a lot of people rely on. I don't think the need to have spaces in an argument to the pager comes up a lot.

Reply:
Ah, good point, thank you. Yeah, this would be much trickier to support, I guess. So, not today...